cdisc_rules_engine/models/dictionaries/abstract_factory.py

```diff
@@ -0,0 +1,37 @@
+from typing import Type
+
+from cdisc_rules_engine.models.dictionaries import (
+    DictionaryTypes,
+    TermsFactoryInterface,
+)
+from cdisc_rules_engine.models.dictionaries.meddra import MedDRATermsFactory
+from cdisc_rules_engine.models.dictionaries.whodrug import WhoDrugTermsFactory
+from cdisc_rules_engine.services.data_services import BaseDataService
+from cdisc_rules_engine.services.factory_interface import FactoryInterface
+
+
+class AbstractTermsFactory(FactoryInterface):
+    _service_map: dict = {
+        DictionaryTypes.MEDDRA.value: MedDRATermsFactory,
+        DictionaryTypes.WHODRUG.value: WhoDrugTermsFactory,
+    }
+
+    def __init__(self, data_service: BaseDataService):
+        self.data_service = data_service
+
+    @classmethod
+    def register_service(cls, name: str, service: Type[TermsFactoryInterface]) -> None:
```

Comment:
These are factories, so it might be more clear to name these functions: register_factory, get_factory and the map: _factory_map

Reply:
I suggest to name it like`_registered_services_map` because _factory_map and _service_map don't describe field purpose as it should. What do you think?

Reply:
That works for me